pkg/tsdb/elasticsearch/client/client.go

```diff
@@ -191,8 +202,14 @@ func (c *baseClientImpl) ExecuteMultisearch(r *MultiSearchRequest) (*MultiSearch
 	start = time.Now()
 	var msr MultiSearchResponse
 	dec := json.NewDecoder(res.Body)
+	_, resSpan := c.tracer.Start(c.ctx, "datasource.elasticsearch.queryData.executeMultisearch.decodeResponse")
+	defer resSpan.End()
 	err = dec.Decode(&msr)
 	if err != nil {
+		span.RecordError(err)
+		span.SetStatus(codes.Error, err.Error())
```

Comment:
Nit: When dealing with multiple sub-spans in a single function, rather than having to set the error for the parent span here, it may be easier/cleaner to deal with errors this way:

```go
// ...

var err error

_, span := c.tracer.Start(c.ctx, "...")

// set the error once before returning, saves you from
// having to remember to call RecordError and SetStatus
// for the outer span on every branch
defer func(){
	if err != nil {
		span.RecordError(err)
		span.SetStatus(codes.Error, err.Error())
	}
	span.End()
}()

// ...

if err = something(); err != nil {
	return err
}
if err = another(); err != nil {
	return err
}

// ...

_, resSpan := c.tracer.Start(c.ctx, "...")
defer func() {
	if err != nil {
		resSpan.RecordError(err)
		resSpan.SetStatus(codes.Error, err.Error())
	}
	resSpan.End()
}()

err = dec.Decode(&msr)
if err != nil {
	return err
}
```

Reply:
Thanks for recommendation! Updated in  https://github.com/grafana/grafana/pull/74750/commits/b8030060194f0b145ffe9fa5c1afc5c74937a276.